### ISSUE TITLE
referenceBlock with group attribute not allowed in catalog_product_view.xml 

### ISSUE THREAD

**Issue by berbua** [magento/magento2#24656]
I would like to change group attribute for certain referenceBlocks in _Magento_Catalog/layout/catalog_product_view.xml_. It is causing exceptions during XML validation.

### Preconditions (*)
1. Magento 2.3.2 & 2.3-develop

### Steps to reproduce (*)
1. Go to _Magento_Catalog/layout/catalog_product_view.xml_
2. Change group for below mentioned reference blocks:
```
<referenceBlock name="product.info.description" group="column_left">
    <arguments>
        <argument name="title" translate="true" xsi:type="string">Description</argument>
        <argument name="contain_content" xsi:type="boolean">false</argument>
        <argument name="sort_order" xsi:type="string">10</argument>
    </arguments>
</referenceBlock>

<referenceBlock name="product.attributes" group="column_right">
    <arguments>
        <argument name="contain_content" xsi:type="boolean">false</argument>
        <argument name="sort_order" xsi:type="string">20</argument>
    </arguments>
</referenceBlock>

<referenceBlock name="catalog.product.related" group="detailed_info">
    <arguments>
        <argument translate="true" name="title" xsi:type="string">Related products</argument>
        <argument name="sort_order" xsi:type="string">80</argument>
        <argument name="contain_content" xsi:type="boolean">false</argument>
        <argument name="include_headline" xsi:type="boolean">false</argument>
        <argument name="is_collapsible" xsi:type="boolean">false</argument>
        <argument name="view_mode" xsi:type="string">grid</argument>
    </arguments>
</referenceBlock>

<referenceBlock name="product.info.upsell" group="detailed_info">
    <arguments>
        <argument translate="true" name="title" xsi:type="string">You may be also interested in following products</argument>
        <argument name="sort_order" xsi:type="string">100</argument>
        <argument name="contain_content" xsi:type="boolean">false</argument>
        <argument name="include_headline" xsi:type="boolean">false</argument>
        <argument name="is_collapsible" xsi:type="boolean">false</argument>
        <argument name="in_nav" xsi:type="boolean">false</argument>
        <argument name="view_mode" xsi:type="string">grid</argument>
    </arguments>
</referenceBlock>
```

3. Go to Product Desciption Page in Developer Mode

### Expected result (*)

1. No errors thrown

### Actual result (*)

1. Exceptions:
```
1 exception(s):
Exception #0 (Magento\Framework\Config\Dom\ValidationException): Element 'referenceBlock', attribute 'group': The attribute 'group' is not allowed.
Line: 2883

Element 'referenceBlock', attribute 'group': The attribute 'group' is not allowed.
Line: 2890

Element 'referenceBlock', attribute 'group': The attribute 'group' is not allowed.
Line: 2915

Element 'referenceBlock', attribute 'group': The attribute 'group' is not allowed.
Line: 2924


Exception #0 (Magento\Framework\Config\Dom\ValidationException): Element 'referenceBlock', attribute 'group': The attribute 'group' is not allowed.
Line: 2883

Element 'referenceBlock', attribute 'group': The attribute 'group' is not allowed.
Line: 2890

Element 'referenceBlock', attribute 'group': The attribute 'group' is not allowed.
Line: 2915

Element 'referenceBlock', attribute 'group': The attribute 'group' is not allowed.
Line: 2924

#0 /var/www/vendor/magento/framework/Config/Dom.php(117): Magento\Framework\Config\Dom->_initDom('<layout xmlns:x...')
```

I tried adding _group_ attribute to `<move>` tag but it also caused errors.
How can _group_ attribute be changed for a referenceBlock?

**Comment by jmcharadva** [comment]
@berbua 
In Magento, there are two attributes "before" and "after" you can use that. 
If you want to move any block to another place then there is 
`<move element="{element-name}" destination="{desitnation}" before="-"/>`

Please try this, it might help you.

**Comment by krzksz** [comment]
This is not what we are trying to achieve. Currently it seems that there is no way allowed by XML validator to change the value of `group` attribute for already defined block using `referenceBlock`.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-20290` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by krisdante** [comment]
Does it work if you ignore validator warning?

**Comment by berbua** [comment]
@krisdante yes, it is working

**Comment by tim-breitenstein-it** [comment]
Still exists on Magento 2.4.2

**Comment by magento-engcom-team** [comment]
Hi @berbua, @korostii. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-24223` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.

**Comment by fritzmg** [comment]
This issue still exists in Magento `2.4.3`. As far as I can see the changes of f8d88d65a716854be69c1f155884a30e89cba160 did not make it into the `2.4.3` release: https://github.com/magento/magento2/blob/4c36116dcf878e127059d9be9566a119783583f2/lib/internal/Magento/Framework/View/Layout/etc/elements.xsd#L330-L331

Is there any specific reason for this?
